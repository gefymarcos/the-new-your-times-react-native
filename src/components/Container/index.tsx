import * as React from 'react'
import { View, StyleSheet, ViewPropTypes } from 'react-native'

type containerProps = {
  children: React.ReactNode
  alignContent?: 'center' | 'top'
}
const Container = ({ children, alignContent = 'top' }: containerProps) => {
  return <View style={[styles.content, styles[alignContent]]}>{children}</View>
}

const styles = StyleSheet.create({
  content: {
    flex: 1,
    paddingTop: 20
  },
  center: {
    justifyContent: 'flex-end',
    alignItems: 'center'
  },
  top: {
    justifyContent: 'flex-start'
  }
})

export default Container
