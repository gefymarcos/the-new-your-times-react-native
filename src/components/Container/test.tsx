import React from 'react'
import { Text } from 'react-native'
import renderer from 'react-test-renderer'
import { render } from '@testing-library/react-native'

import Container from '.'

jest.useFakeTimers()

describe('<Tabs />', () => {
  it('should render correctly', () => {
    const { getByText } = render(
      <Container>
        <Text>teste</Text>
      </Container>
    )

    const content = getByText(/teste/i)

    expect(content).toBeDefined()
  })

  it('should render correctly based on snapshot', () => {
    const tree = renderer
      .create(
        <Container>
          <Text>teste</Text>
        </Container>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
