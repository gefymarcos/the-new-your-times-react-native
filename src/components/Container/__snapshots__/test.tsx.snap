// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tabs /> should render correctly based on snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "paddingTop": 20,
      },
      Object {
        "justifyContent": "flex-start",
      },
    ]
  }
>
  <Text>
    teste
  </Text>
</View>
`
